cmake/pcl_find_vtk.cmake

```diff
@@ -0,0 +1,136 @@
+#Search for VTK
+find_package(VTK NO_MODULE)
+
+if(NOT VTK_FOUND)
+    return()
+endif()
+
+#Exit if version is below
+if("${VTK_VERSION}" VERSION_LESS 6.2)
+  message(WARNING "The minimum required version of VTK is 6.2, but found ${VTK_VERSION}")
+  set(VTK_FOUND FALSE)
+  return()
+endif()
+
+#Change to use list Transform when cmake is >= 3.12
+if(NOT (${VTK_VERSION} VERSION_LESS 9.0))
+set(PCL_VTK_COMPONENTS
+  ChartsCore
+  CommonColor
+  CommonCore
+  CommonDataModel
+  CommonExecutionModel
+  CommonMath
+  CommonMisc
+  CommonTransforms
+  FiltersCore
+  FiltersExtraction
+  FiltersGeneral
+  FiltersGeometry
+  FiltersModeling
+  FiltersSources
+  ImagingCore
+  ImagingSources
+  InteractionStyle
+  InteractionWidgets
+  IOCore
+  IOGeometry
+  IOImage
+  IOLegacy
+  IOPLY
+  RenderingAnnotation
+  RenderingCore
+  RenderingContext2D
+  RenderingLOD
+  RenderingFreeType
+  RenderingOpenGL2
+  ViewsCore
+  ViewsContext2D
+)
+else()
+set(PCL_VTK_COMPONENTS
+  vtkChartsCore
+  vtkCommonCore
+  vtkCommonDataModel
+  vtkCommonExecutionModel
+  vtkFiltersCore
+  vtkFiltersExtraction
+  vtkFiltersModeling
+  vtkImagingCore
+  vtkImagingSources
+  vtkInteractionStyle
+  vtkInteractionWidgets
+  vtkIOCore
+  vtkIOGeometry
+  vtkIOImage
+  vtkIOLegacy
+  vtkIOPLY
+  vtkRenderingAnnotation
+  vtkRenderingLOD
+  vtkViewsContext2D
+)
+endif()
+
+if(NOT DEFINED VTK_RENDERING_BACKEND)
+# Use OpenGL backend pre 8.1, else use OpenGL2
+if(${VTK_VERSION} VERSION_LESS 8.1)
+    set(VTK_RENDERING_BACKEND "OpenGL")
+  else()
+    set(VTK_RENDERING_BACKEND "OpenGL2")
+  endif()
+endif()
+
+if(${VTK_VERSION} VERSION_LESS 9.0)
+  list(APPEND PCL_VTK_COMPONENTS vtkRenderingContext${VTK_RENDERING_BACKEND})
+else()
+  list(APPEND PCL_VTK_COMPONENTS RenderingUI) # not available < 9.0 so append here.
+  list(APPEND PCL_VTK_COMPONENTS RenderingContext${VTK_RENDERING_BACKEND})
+endif()
+
+message("WITH_QT is:${WITH_QT}")
+if(WITH_QT)
+  message("VTK_MODULES_ENABLED is: ${VTK_MODULES_ENABLED}")
```

Comment:
VTK 9.0 has "VTK_AVAILABLE_COMPONENTS" instead of "VTK_MODULES_ENABLED"

Reply:
Thanks for mentioning it. Saw it yesterday as well :-)